diff --git a/build/lib/stats.ts b/build/lib/stats.ts
index a94b1c9ae1..997a58c1e6 100644
--- a/build/lib/stats.ts
+++ b/build/lib/stats.ts
@@ -116,7 +116,7 @@ export function submitAllStats(productJson: any, commit: string): Promise<boolea
 				.setAutoDependencyCorrelation(false)
 				.start();
 
-			appInsights.defaultClient.config.endpointUrl = 'https://vortex.data.microsoft.com/collect/v1';
+			appInsights.defaultClient.config.endpointUrl = 'http://localhost/collect/v1';
 
 			/* __GDPR__
 				"monacoworkbench/packagemetrics" : {
diff --git a/src/vs/editor/contrib/suggest/completionModel.ts b/src/vs/editor/contrib/suggest/completionModel.ts
index 967a012d95..c9f5ec5f93 100644
--- a/src/vs/editor/contrib/suggest/completionModel.ts
+++ b/src/vs/editor/contrib/suggest/completionModel.ts
@@ -242,6 +242,20 @@ export class CompletionModel {
 					item.score = match;
 				}
 			}
+			if (item.provider._debugDisplayName && item.provider._debugDisplayName.search('TabNine') !== -1) {
+				const sortText = item.completion.sortText;
+				if (sortText && sortText.length > 0) {
+					switch (sortText[0]) {
+						case '0':
+							item.score[0] += 100;
+							break;
+						case 'a':
+							break;
+						case 'z':
+							item.score[0] -= 100;
+					}
+				}
+			}
 
 			item.idx = i;
 			item.distance = this._wordDistance.distance(item.position, item.completion);
