--- a/runtime/grammars/sources/rescript/src/scanner.c
+++ b/runtime/grammars/sources/rescript/src/scanner.c
@@ -131,7 +131,7 @@ bool tree_sitter_rescript_external_scanner_scan(
     const bool* valid_symbols
     ) {
   ScannerState* state = (ScannerState*)payload;
-  const in_string = state->in_quotes || state->in_backticks;
+  const bool in_string = state->in_quotes || state->in_backticks;
 
   if (valid_symbols[TEMPLATE_CHARS]) {
     lexer->result_symbol = TEMPLATE_CHARS;
--- a/runtime/grammars/sources/d/src/scanner.c
+++ b/runtime/grammars/sources/d/src/scanner.c
@@ -7,6 +7,7 @@
  * (See accompanying file LICENSE.txt or https://opensource.org/licenses/MIT)
  * SPDX-License-Identifier: MIT
  */
+#define _POSIX_SOURCE
 #include "tree_sitter/parser.h"
 #include <assert.h>
 #include <ctype.h>
