diff --git a/include/nanobind/nb_backend.h b/include/nanobind/nb_backend.h
index 5732df5..c20de1a 100644
--- a/include/nanobind/nb_backend.h
+++ b/include/nanobind/nb_backend.h
@@ -38,6 +38,8 @@
     BSD-style license that can be found in the LICENSE file.
 */
 
+#include <stdio.h>
+
 /// Major version of the backend ABI. Advances after ABI-breaking backend
 /// changes. Such changes are to be avoided at all costs.
 #define NB_BACKEND_ABI_MAJOR 1
