diff --git a/libobs/util/platform-nix.c b/libobs/util/platform-nix.c
index d429d99..3fdab36 100644
--- a/libobs/util/platform-nix.c
+++ b/libobs/util/platform-nix.c
@@ -73,8 +73,7 @@ void *os_dlopen(const char *path)
 #ifdef __APPLE__
 	void *res = dlopen(dylib_name.array, RTLD_LAZY | RTLD_FIRST);
 #else
-	void *res = dlopen(dylib_name.array,
-			   RTLD_LAZY | RTLD_DEEPBIND | RTLD_GLOBAL);
+	void *res = dlopen(dylib_name.array, RTLD_LAZY | RTLD_GLOBAL);
 #endif
 	if (!res)
 		blog(LOG_ERROR, "os_dlopen(%s->%s): %s\n", path,
