diff --git a/tool/transform_mjit_header.rb b/tool/transform_mjit_header.rb
index 2180236..5774dd2 100644
--- a/tool/transform_mjit_header.rb
+++ b/tool/transform_mjit_header.rb
@@ -216,7 +216,7 @@ end
 cc      = ARGV[0]
 code    = File.binread(ARGV[1]) # Current version of the header file.
 outfile = ARGV[2]
-cflags = '-S -DMJIT_HEADER -fsyntax-only -Werror=implicit-function-declaration -Werror=implicit-int -Wfatal-errors'
+cflags = '-std=gnu2x -S -DMJIT_HEADER -fsyntax-only -Werror=implicit-function-declaration -Werror=implicit-int -Wfatal-errors'
 
 if !MJITHeader.supported_header?(code)
   puts "This compiler (#{cc}) looks not supported for MJIT. Giving up to generate MJIT header."
