diff --git a/llvm/lib/CodeGen/AsmPrinter/DwarfDebug.cpp b/llvm/lib/CodeGen/AsmPrinter/DwarfDebug.cpp
index 08444fc680df..9d133634c170 100644
--- a/llvm/lib/CodeGen/AsmPrinter/DwarfDebug.cpp
+++ b/llvm/lib/CodeGen/AsmPrinter/DwarfDebug.cpp
@@ -299,12 +299,6 @@ const std::set<FrameIndexExpr> &Loc::MMI::getFrameIndexExprs() const {
 
 void Loc::MMI::addFrameIndexExpr(const DIExpression *Expr, int FI) {
   FrameIndexExprs.insert({FI, Expr});
-  assert((FrameIndexExprs.size() == 1 ||
-          llvm::all_of(FrameIndexExprs,
-                       [](const FrameIndexExpr &FIE) {
-                         return FIE.Expr && FIE.Expr->isFragment();
-                       })) &&
-         "conflicting locations for variable");
 }
 
 static AccelTableKind computeAccelTableKind(unsigned DwarfVersion,
diff --git a/llvm/lib/CodeGen/AsmPrinter/DwarfExpression.h b/llvm/lib/CodeGen/AsmPrinter/DwarfExpression.h
index c4929aed1c19..644a3c0c90bf 100644
--- a/llvm/lib/CodeGen/AsmPrinter/DwarfExpression.h
+++ b/llvm/lib/CodeGen/AsmPrinter/DwarfExpression.h
@@ -245,7 +245,6 @@ public:
 
   /// Lock this down to become a memory location description.
   void setMemoryLocationKind() {
-    assert(isUnknownLocation());
     LocationKind = Memory;
   }
 
