diff --git a/llvm/lib/CodeGen/AsmPrinter/DwarfDebug.cpp b/llvm/lib/CodeGen/AsmPrinter/DwarfDebug.cpp
index 08444fc680df..9d133634c170 100644
--- a/llvm/lib/CodeGen/AsmPrinter/DwarfDebug.cpp
+++ b/llvm/lib/CodeGen/AsmPrinter/DwarfDebug.cpp
@@ -299,12 +299,6 @@ const std::set<FrameIndexExpr> &Loc::MMI::getFrameIndexExprs() const {
 
 void Loc::MMI::addFrameIndexExpr(const DIExpression *Expr, int FI) {
   FrameIndexExprs.insert({FI, Expr});
-  assert((FrameIndexExprs.size() == 1 ||
-          llvm::all_of(FrameIndexExprs,
-                       [](const FrameIndexExpr &FIE) {
-                         return FIE.Expr && FIE.Expr->isFragment();
-                       })) &&
-         "conflicting locations for variable");
 }
 
 static AccelTableKind computeAccelTableKind(unsigned DwarfVersion,
