diff --git a/libxtables/xtables.c b/libxtables/xtables.c
index 96fd783..f003a3f 100644
--- a/libxtables/xtables.c
+++ b/libxtables/xtables.c
@@ -2245,8 +2245,6 @@ void xtables_print_num(uint64_t number, unsigned int format)
 	printf(FMT("%4lluT ","%lluT "), (unsigned long long)number);
 }
 
-#include <netinet/ether.h>
-
 static const unsigned char mac_type_unicast[ETH_ALEN] =   {};
 static const unsigned char msk_type_unicast[ETH_ALEN] =   {1};
 static const unsigned char mac_type_multicast[ETH_ALEN] = {1};
