diff --git a/lib/external/libromfs/lib/include/romfs/romfs.hpp b/lib/external/libromfs/lib/include/romfs/romfs.hpp
index 2f4c90b..f07d0e5 100644
--- a/lib/external/libromfs/lib/include/romfs/romfs.hpp
+++ b/lib/external/libromfs/lib/include/romfs/romfs.hpp
@@ -6,6 +6,7 @@
 #include <string_view>
 #include <span>
 #include <vector>
+#include <type_traits>
 
 #define ROMFS_CONCAT_IMPL(x, y) x##y
 #define ROMFS_CONCAT(x, y) ROMFS_CONCAT_IMPL(x, y)
diff --git a/plugins/builtin/source/content/providers/disk_provider.cpp b/plugins/builtin/source/content/providers/disk_provider.cpp
index 5a96873..9e13f33 100644
--- a/plugins/builtin/source/content/providers/disk_provider.cpp
+++ b/plugins/builtin/source/content/providers/disk_provider.cpp
@@ -44,7 +44,7 @@
 #endif
 
 #if defined(OS_LINUX) && !defined(OS_FREEBSD)
-    #define lseek lseek64
+    //#define lseek lseek64
 #elif defined(OS_FREEBSD)
     #include <sys/disk.h>
     #define DEFAULT_SECTOR_SIZE 512
