diff --git a/src/reshade/source/effect_codegen.hpp b/src/reshade/source/effect_codegen.hpp
index 1e278075..572c9bd9 100644
--- a/src/reshade/source/effect_codegen.hpp
+++ b/src/reshade/source/effect_codegen.hpp
@@ -339,7 +339,7 @@ namespace reshadefx
 			return align_up(size, alignment) * (elements - 1) + size;
 		}
 
-		module _module;
+		struct module _module;
 		std::vector<struct_info> _structs;
 		std::vector<std::unique_ptr<function_info>> _functions;
 		id _next_id = 1;
diff --git a/src/reshade/source/effect_codegen_spirv.cpp b/src/reshade/source/effect_codegen_spirv.cpp
index 57190055..e3e82b20 100644
--- a/src/reshade/source/effect_codegen_spirv.cpp
+++ b/src/reshade/source/effect_codegen_spirv.cpp
@@ -6,6 +6,7 @@
 #include "effect_parser.hpp"
 #include "effect_codegen.hpp"
 #include <cassert>
+#include <cstdlib> // std::strtoul
 #include <cstring> // memcmp
 #include <algorithm> // std::find_if, std::max
 #include <unordered_set>
diff --git a/src/reshade/source/effect_symbol_table.cpp b/src/reshade/source/effect_symbol_table.cpp
index 3a84c6c1..47dbb704 100644
--- a/src/reshade/source/effect_symbol_table.cpp
+++ b/src/reshade/source/effect_symbol_table.cpp
@@ -6,6 +6,7 @@
 #include "effect_symbol_table.hpp"
 #include <cassert>
 #include <malloc.h> // alloca
+#include <alloca.h>
 #include <algorithm> // std::upper_bound, std::sort
 #include <functional> // std::greater
 #include <span>
