diff --git a/core/src/librscpp/poly.h b/core/src/librscpp/poly.h
index 4beb5ff..a08b5d1 100644
--- a/core/src/librscpp/poly.h
+++ b/core/src/librscpp/poly.h
@@ -10,12 +10,13 @@
 #include <numeric>
 #include <span>
 #include <vector>
+#include <cmath>
 
 namespace librscpp {
 
 /**
  * @brief Represents a polynomial whose coefficients are elements of Field.
- * 
+ *
  * The coefficients are stored in a vector, arranged from most significant (highest-power term) to least significant.
  */
 template <typename Field>
