diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4f41543..c59573d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -131,7 +131,7 @@ if(APPLE)
     set(CMAKE_MACOSX_RPATH 1)
 endif()
 
-option(BUILD_WERROR "Treat compiler warnings as errors" ON)
+option(BUILD_WERROR "Treat compiler warnings as errors" OFF)
 if(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "Clang")
     set(COMMON_COMPILE_FLAGS "-Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers")
     set(COMMON_COMPILE_FLAGS "${COMMON_COMPILE_FLAGS} -fno-strict-aliasing -fno-builtin-memcmp")
