diff --git a/src/rcw.c b/src/rcw.c
index 45e2277..1183924 100644
--- a/src/rcw.c
+++ b/src/rcw.c
@@ -3565,11 +3565,12 @@ static GtkWidget *rco_create_tab_label(RemminaConnectionObject *cnnobj)
 	gtk_widget_show(widget);
 	gtk_box_pack_start(GTK_BOX(hbox), widget, FALSE, FALSE, 0);
 
-	gchar* label = remmina_file_get_string(cnnobj->remmina_file, "name");
+	gchar* label = g_strdup(remmina_file_get_string(cnnobj->remmina_file, "name"));
 	if (strlen(label) > 100){
 		label[99] = 0;
 	}
 	widget = gtk_label_new(label);
+	g_free(label);
 	gtk_widget_set_valign(widget, GTK_ALIGN_CENTER);
 	gtk_widget_set_halign(widget, GTK_ALIGN_CENTER);
 
diff --git a/src/remmina_exec.c b/src/remmina_exec.c
index 21181c1..812cfc8 100644
--- a/src/remmina_exec.c
+++ b/src/remmina_exec.c
@@ -284,7 +284,7 @@ static void remmina_exec_connect(const gchar *data)
 
 	protocolserver = g_strsplit(data, "://", 2);
 	if (strncmp(protocol, "WWW", 3) == 0){
-		server = data;
+		server = g_strdup(data);
 	}
 	else{
 		server = g_strdup(protocolserver[1]);
