--- a/libunwind/src/libunwind.cpp
+++ b/;ibunwind/src/libunwind.cpp
@@ -297,7 +297,7 @@ void __unw_add_dynamic_fde(unw_word_t fde) {
                                           fdeInfo.pcStart, fdeInfo.pcEnd,
                                           fdeInfo.fdeStart);
   } else {
-    _LIBUNWIND_DEBUG_LOG("__unw_add_dynamic_fde: bad fde: %s", message);
+    //_LIBUNWIND_DEBUG_LOG("__unw_add_dynamic_fde: bad fde: %s", message);
   }
 }
 
