diff --git a/tools/winegcc/winegcc.c b/tools/winegcc/winegcc.c
index 0e9ad6b..7652e27 100644
--- a/tools/winegcc/winegcc.c
+++ b/tools/winegcc/winegcc.c
@@ -435,6 +435,7 @@ static struct strarray build_tool_name( const char *target_name, struct tool_nam
         }
         strarray_add( &ret, "-Wno-unused-command-line-argument" );
         strarray_add( &ret, "-fuse-ld=lld" );
+        strarray_add( &ret, "--rtlib=compiler-rt" );
         if (no_default_config) strarray_add( &ret, "--no-default-config" );
     }
     return ret;
