diff --git a/sal/osl/unx/file.cxx b/sal/osl/unx/file.cxx
index ee7412dd1..f36f5381d 100644
--- a/sal/osl/unx/file.cxx
+++ b/sal/osl/unx/file.cxx
@@ -67,6 +67,9 @@
 #ifdef LINUX
 #include <sys/vfs.h>
 // As documented by the kernel
+#ifndef __fsword_t
+#define __fsword_t unsigned int
+#endif
 #define SMB_SUPER_MAGIC  static_cast<__fsword_t>(0x517B)
 #define CIFS_SUPER_MAGIC static_cast<__fsword_t>(0xFF534D42)
 #define SMB2_SUPER_MAGIC static_cast<__fsword_t>(0xFE534D42)
