diff --git a/src/amd/common/ac_rtld.c b/src/amd/common/ac_rtld.c
index 4b54da4..80fca95 100644
--- a/src/amd/common/ac_rtld.c
+++ b/src/amd/common/ac_rtld.c
@@ -650,6 +650,9 @@ static bool apply_relocs(const struct ac_rtld_upload_info *u, unsigned part_idx,
       uint64_t symbol;
       uint64_t addend;
 
+#ifndef STN_UNDEF
+#define STN_UNDEF 0
+#endif
       if (r_sym == STN_UNDEF) {
          symbol = 0;
       } else {
