diff --git a/CMakeLists.txt b/CMakeLists.txt
index f85b618..2b064d0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -182,20 +182,6 @@ else()
     set(CPP_DEFS ${CPP_DEFS} "RESTRICT=")
 endif()
 
-# Some systems may need libatomic for atomic functions to work
-set(OLD_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES})
-set(CMAKE_REQUIRED_LIBRARIES ${OLD_REQUIRED_LIBRARIES} atomic)
-check_cxx_source_compiles("#include <atomic>
-std::atomic<int> foo{0};
-int main() { return foo.fetch_add(2); }"
-HAVE_LIBATOMIC)
-if(NOT HAVE_LIBATOMIC)
-    set(CMAKE_REQUIRED_LIBRARIES "${OLD_REQUIRED_LIBRARIES}")
-else()
-    set(EXTRA_LIBS atomic ${EXTRA_LIBS})
-endif()
-unset(OLD_REQUIRED_LIBRARIES)
-
 # Include liblog for Android logging
 check_library_exists(log __android_log_print "" HAVE_LIBLOG)
 if(HAVE_LIBLOG)
