diff --git a/3rd/bee.lua/compile/common.lua b/3rd/bee.lua/compile/common.lua
index 85c14ea..7409e77 100644
--- a/3rd/bee.lua/compile/common.lua
+++ b/3rd/bee.lua/compile/common.lua
@@ -141,7 +141,6 @@ lm:lua_source "source_bee" {
     },
     linux = {
         sources = "!binding/lua_unicode.cpp",
-        links = "stdc++fs",
         ldflags = "-pthread"
     },
     macos = {
@@ -227,8 +226,5 @@ lm:source_set 'source_lua' {
     },
     msvc = {
         flags = "/wd4267"
-    },
-    gcc = {
-        flags = "-Wno-maybe-uninitialized",
     }
 }
diff --git a/3rd/luamake/bee.lua/compile/common.lua b/3rd/luamake/bee.lua/compile/common.lua
index 85c14ea..ba29048 100644
--- a/3rd/luamake/bee.lua/compile/common.lua
+++ b/3rd/luamake/bee.lua/compile/common.lua
@@ -141,7 +141,6 @@ lm:lua_source "source_bee" {
     },
     linux = {
         sources = "!binding/lua_unicode.cpp",
-        links = "stdc++fs",
         ldflags = "-pthread"
     },
     macos = {
diff --git a/3rd/luamake/compile/ninja/linux.ninja b/3rd/luamake/compile/ninja/linux.ninja
index 326fd61..954bfe8 100644
--- a/3rd/luamake/compile/ninja/linux.ninja
+++ b/3rd/luamake/compile/ninja/linux.ninja
@@ -95,8 +95,8 @@ build $obj/source_bootstrap/main.obj: cxx_source_bootstrap $
 build $obj/source_bootstrap/progdir.obj: cxx_source_bootstrap $
     bee.lua/bootstrap/progdir.cpp
 rule link_bootstrap
-  command = $cc $in -o $out -Wl,-E -static-libgcc -lm -ldl -lstdc++fs $
-    -pthread -Wl,-Bstatic -lstdc++ -Wl,-Bdynamic LDFLAGS
+  command = $cc $in -o $out -Wl,-E -static-libgcc -lm -ldl -lc++ $
+    -pthread -Wl,-Bstatic -Wl,-Bdynamic LDFLAGS
   description = Link    Exe $out
 build $bin/bootstrap: link_bootstrap $obj/source_bootstrap/main.obj $
     $obj/source_bootstrap/progdir.obj $obj/source_bee/lua-seri.obj $
