From e9d9014421386cfe7016859a2c714164b2452c4a Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Mon, 6 Apr 2026 20:28:14 +0800
Subject: [PATCH] Fix behavior of --omagic

---
 src/passes.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/passes.cc b/src/passes.cc
index 2ad37b7c..2f47edea 100644
--- a/src/passes.cc
+++ b/src/passes.cc
@@ -2731,7 +2731,7 @@ static void set_virtual_addresses_regular(Context<E> &ctx) {
       i64 flags1 = get_flags(chunks[i - 1]);
       i64 flags2 = get_flags(chunks[i]);
 
-      if (!ctx.arg.nmagic && flags1 != flags2) {
+      if (!ctx.arg.nmagic && !ctx.arg.omagic && flags1 != flags2) {
         switch (ctx.arg.z_separate_code) {
         case SEPARATE_LOADABLE_SEGMENTS:
           addr = align_to(addr, ctx.page_size);
-- 
2.53.0

