diff --git a/src/output-chunks.cc b/src/output-chunks.cc
index bfbdefb9..f837838b 100644
--- a/src/output-chunks.cc
+++ b/src/output-chunks.cc
@@ -695,6 +695,7 @@ static std::vector<Word<E>> create_dynamic_section(Context<E> &ctx) {
 
   if (ctx.dynsym->shdr.sh_size) {
     define(DT_SYMTAB, ctx.dynsym->shdr.sh_addr);
+    define(DT_SYMTABSZ, ctx.dynsym->shdr.sh_size);
     define(DT_SYMENT, sizeof(ElfSym<E>));
   }
 
diff --git a/src/elf.h b/src/elf.h
index fe42701..6ed54cc 100644
--- a/src/elf.h
+++ b/src/elf.h
@@ -287,6 +287,7 @@ enum : u32 {
   DT_RELRSZ = 35,
   DT_RELR = 36,
   DT_RELRENT = 37,
+  DT_SYMTABSZ = 39,
   DT_ANDROID_REL = 0x6000000f,
   DT_ANDROID_RELSZ = 0x60000010,
   DT_ANDROID_RELA = 0x60000011,
