From d4ba1f20d80b5d6847c79137bc81bc3298b71b99 Mon Sep 17 00:00:00 2001
From: 12101111 <w12101111@gmail.com>
Date: Tue, 14 Nov 2023 17:11:35 +0800
Subject: [PATCH 1/2] removed in 80077787f8f21da1efd8dc27a4c5767ab47a1df6

---
 include/libnftnl/chain.h | 4 ----
 src/libnftnl.map         | 2 --
 2 files changed, 6 deletions(-)

diff --git a/include/libnftnl/chain.h b/include/libnftnl/chain.h
index f56e581..bac1f5f 100644
--- a/include/libnftnl/chain.h
+++ b/include/libnftnl/chain.h
@@ -71,10 +71,6 @@ struct nlmsghdr;
 
 void nftnl_chain_nlmsg_build_payload(struct nlmsghdr *nlh, const struct nftnl_chain *t);
 
-int nftnl_chain_parse(struct nftnl_chain *c, enum nftnl_parse_type type,
-		    const char *data, struct nftnl_parse_err *err);
-int nftnl_chain_parse_file(struct nftnl_chain *c, enum nftnl_parse_type type,
-			 FILE *fp, struct nftnl_parse_err *err);
 int nftnl_chain_snprintf(char *buf, size_t size, const struct nftnl_chain *t, uint32_t type, uint32_t flags);
 int nftnl_chain_fprintf(FILE *fp, const struct nftnl_chain *c, uint32_t type, uint32_t flags);
 
diff --git a/src/libnftnl.map b/src/libnftnl.map
index ad8f2af..64ed559 100644
--- a/src/libnftnl.map
+++ b/src/libnftnl.map
@@ -47,8 +47,6 @@ global:
   nftnl_chain_get_s32;
   nftnl_chain_get_u64;
   nftnl_chain_get_str;
-  nftnl_chain_parse;
-  nftnl_chain_parse_file;
   nftnl_chain_snprintf;
   nftnl_chain_fprintf;
   nftnl_chain_nlmsg_build_payload;
-- 
2.42.1

