diff --git a/lib/utils_math.c b/lib/utils_math.c
index a7e7474..4421d5b 100644
--- a/lib/utils_math.c
+++ b/lib/utils_math.c
@@ -5,6 +5,7 @@
 #include <string.h>
 #include <math.h>
 #include <asm/types.h>
+#include <limits.h>
 
 #include "utils.h"
 
diff --git a/tc/tc_core.c b/tc/tc_core.c
index 63d317c..0f787c2 100644
--- a/tc/tc_core.c
+++ b/tc/tc_core.c
@@ -15,6 +15,7 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <string.h>
+#include <limits.h>
 
 #include "utils.h"
 #include "tc_core.h"
