--- a/configure	2020-05-12 17:27:04.754030693 +0800
+++ b/configure	2020-05-12 18:53:20.335188639 +0800
@@ -5702,7 +5702,11 @@
   return 0;
 }
 EOF
-  if compile_prog "" "" ; then
+  if compile_prog "" "$LIBS" ; then
+    atomic128=yes
+  elif compile_prog "" "$LIBS -latomic" ; then
+    LIBS="$LIBS -latomic"
+    libs_qga="$libs_qga -latomic"
     atomic128=yes
   fi
 fi
@@ -5717,7 +5721,11 @@
   return 0;
 }
 EOF
-  if compile_prog "" "" ; then
+  if compile_prog "" "$LIBS" ; then
+    cmpxchg128=yes
+  elif compile_prog "" "$LIBS-latomic" ; then
+    LIBS="$LIBS -latomic"
+    libs_qga="$libs_qga -latomic"
     cmpxchg128=yes
   fi
 fi
@@ -5747,10 +5755,15 @@
   return 0;
 }
 EOF
-if compile_prog "" "" ; then
+if compile_prog "" "$LIBS" ; then
+  atomic64=yes
+elif compile_prog "" "$LIBS -latomic" ; then
+  LIBS="$LIBS -latomic"
+  libs_qga="$libs_qga -latomic"
   atomic64=yes
 fi
 
+
 #########################################
 # See if --dynamic-list is supported by the linker
 ld_dynamic_list="no"
