--- a/util/bufferiszero.c	2020-04-29 00:49:25.000000000 +0800
+++ b/util/bufferiszero.c	2020-05-12 17:18:27.693014913 +0800
@@ -68,6 +68,7 @@
  * does not support them.
  */
 #if defined(CONFIG_AVX512F_OPT) || defined(CONFIG_AVX2_OPT)
+#pragma clang attribute push (__attribute__((target("sse2"))), apply_to=function)
 #pragma GCC push_options
 #pragma GCC target("sse2")
 #endif
@@ -105,6 +106,7 @@
     return _mm_movemask_epi8(_mm_cmpeq_epi8(t, zero)) == 0xFFFF;
 }
 #if defined(CONFIG_AVX512F_OPT) || defined(CONFIG_AVX2_OPT)
+#pragma clang attribute pop
 #pragma GCC pop_options
 #endif
 
@@ -113,6 +115,7 @@
  * the includes have to be within the corresponding push_options region, and
  * therefore the regions themselves have to be ordered with increasing ISA.
  */
+#pragma clang attribute push (__attribute__((target("sse4"))), apply_to=function)
 #pragma GCC push_options
 #pragma GCC target("sse4")
 #include <smmintrin.h>
@@ -145,7 +148,10 @@
     return _mm_testz_si128(t, t);
 }
 
+#pragma clang attribute pop
 #pragma GCC pop_options
+
+#pragma clang attribute push (__attribute__((target("avx2"))), apply_to=function)
 #pragma GCC push_options
 #pragma GCC target("avx2")
 #include <immintrin.h>
@@ -176,10 +182,12 @@
 
     return _mm256_testz_si256(t, t);
 }
+#pragma clang attribute pop
 #pragma GCC pop_options
 #endif /* CONFIG_AVX2_OPT */
 
 #ifdef CONFIG_AVX512F_OPT
+#pragma clang attribute push (__attribute__((target("avx512f"))), apply_to=function)
 #pragma GCC push_options
 #pragma GCC target("avx512f")
 #include <immintrin.h>
@@ -210,6 +218,7 @@
     return !_mm512_test_epi64_mask(t, t);
 
 }
+#pragma clang attribute pop
 #pragma GCC pop_options
 #endif
 
--- a/configure	2020-05-12 16:41:06.393946514 +0800
+++ b/configure	2020-05-12 17:22:14.044021821 +0800
@@ -5618,6 +5618,7 @@
 
 if test "$cpuid_h" = "yes" && test "$avx2_opt" != "no"; then
   cat > $TMPC << EOF
+#pragma clang attribute push (__attribute__((target("avx2"))), apply_to=function)
 #pragma GCC push_options
 #pragma GCC target("avx2")
 #include <cpuid.h>
@@ -5627,6 +5628,7 @@
     return _mm256_testz_si256(x, x);
 }
 int main(int argc, char *argv[]) { return bar(argv[0]); }
+#pragma clang attribute pop
 EOF
   if compile_object "" ; then
     avx2_opt="yes"
@@ -5645,6 +5647,7 @@
 
 if test "$cpuid_h" = "yes" && test "$avx512f_opt" = "yes"; then
   cat > $TMPC << EOF
+#pragma clang attribute push (__attribute__((target("avx512f"))), apply_to=function)
 #pragma GCC push_options
 #pragma GCC target("avx512f")
 #include <cpuid.h>
@@ -5657,6 +5660,7 @@
 {
 	return bar(argv[0]);
 }
+#pragma clang attribute pop
 EOF
   if ! compile_object "" ; then
     avx512f_opt="no"
