diff --git a/linux-user/syscall.c b/linux-user/syscall.c
index 866f70a..4030d52 100644
--- a/linux-user/syscall.c
+++ b/linux-user/syscall.c
@@ -764,6 +764,18 @@ safe_syscall6(ssize_t, copy_file_range, int, infd, loff_t *, pinoff,
  *  use the flock64 struct rather than unsuffixed flock
  * This will then work and use a 64-bit offset for both 32-bit and 64-bit hosts.
  */
+#ifndef F_GETLK64
+#define F_GETLK64 F_GETLK
+#define F_SETLK64 F_SETLK
+#define F_SETLKW64 F_SETLKW
+#define flock64 flock
+#define off64_t off_t
+#define lseek64 lseek
+#define pread64 pread
+#define pwrite64 pwrite
+#define truncate64 truncate
+#define ftruncate64 ftruncate
+#endif
 #ifdef __NR_fcntl64
 #define safe_fcntl(...) safe_syscall(__NR_fcntl64, __VA_ARGS__)
 #else
