diff --git a/src/analyze/gui/gui.cpp b/src/analyze/gui/gui.cpp
index 7197ea1..8d0f620 100644
--- a/src/analyze/gui/gui.cpp
+++ b/src/analyze/gui/gui.cpp
@@ -23,7 +23,7 @@
 #include <QResource>
 
 // FIXME: patch KIconTheme so that this isn't needed here
-void Q_DECL_UNUSED initRCCIconTheme()
+void initRCCIconTheme()
 {
     const QString iconThemeRcc = qApp->applicationDirPath() + QStringLiteral("/../share/icons/breeze/breeze-icons.rcc");
     if (!QFile::exists(iconThemeRcc)) {
