diff --git a/lld/ELF/SyntheticSections.cpp b/lld/ELF/SyntheticSections.cpp
index 06d5129844e5..d9d10698fdc0 100644
--- a/lld/ELF/SyntheticSections.cpp
+++ b/lld/ELF/SyntheticSections.cpp
@@ -1337,6 +1337,7 @@ DynamicSection<ELFT>::computeContents() {
 
   addInSec(DT_SYMTAB, *ctx.in.dynSymTab);
   addInt(DT_SYMENT, sizeof(Elf_Sym));
+  addInt(DT_SYMTABSZ, ctx.in.dynSymTab->getSize());
   addInSec(DT_STRTAB, *ctx.in.dynStrTab);
   addInt(DT_STRSZ, ctx.in.dynStrTab->getSize());
   if (!ctx.arg.zText)
