diff --git a/include/dosbox.h b/include/dosbox.h
index 7054459..69c90c5 100644
--- a/include/dosbox.h
+++ b/include/dosbox.h
@@ -77,6 +77,25 @@
 /* As long as FILE_OFFSET_BITS==64 Linux is quite good at allowing stat to work with 64-bit sizes */
 # define pref_stat			stat
 # define pref_struct_stat	struct stat
+#define F_GETLK64 F_GETLK
+#define F_SETLK64 F_SETLK
+#define F_SETLKW64 F_SETLKW
+#define flock64 flock
+#define open64 open
+#define openat64 openat
+#define creat64 creat
+#define lockf64 lockf
+#define posix_fadvise64 posix_fadvise
+#define posix_fallocate64 posix_fallocate
+#define off64_t off_t
+#define tmpfile64 tmpfile
+#define fopen64 fopen
+#define freopen64 freopen
+#define fseeko64 fseeko
+#define ftello64 ftello
+#define fgetpos64 fgetpos
+#define fsetpos64 fsetpos
+#define fpos64_t fpos_t
 #endif
 
 // TODO: The autoconf script should test the size of long double
