diff --git a/src/app/CMakeLists.txt b/src/app/CMakeLists.txt
index 69e6520..121a53c 100644
--- a/src/app/CMakeLists.txt
+++ b/src/app/CMakeLists.txt
@@ -139,8 +139,6 @@ if (STACKTRACE)
         endif()
 
         target_link_libraries(qbt_app PUBLIC dbghelp)
-    else()
-        target_sources(qbt_app PRIVATE stacktrace.h)
     endif()
 endif()
 
diff --git a/src/app/app.pri b/src/app/app.pri
index 14db0a1..8296523 100644
--- a/src/app/app.pri
+++ b/src/app/app.pri
@@ -16,17 +16,3 @@ SOURCES += \
     $$PWD/main.cpp \
     $$PWD/qtlocalpeer/qtlocalpeer.cpp \
     $$PWD/upgrade.cpp
-
-stacktrace {
-    unix {
-        HEADERS += $$PWD/stacktrace.h
-    }
-    else {
-        HEADERS += $$PWD/stacktrace_win.h
-        !nogui {
-            HEADERS += $$PWD/stacktracedialog.h
-            SOURCES += $$PWD/stacktracedialog.cpp
-            FORMS += $$PWD/stacktracedialog.ui
-        }
-    }
-}
diff --git a/src/app/main.cpp b/src/app/main.cpp
index e1406f9..01858fb 100644
--- a/src/app/main.cpp
+++ b/src/app/main.cpp
@@ -65,17 +65,6 @@ Q_IMPORT_PLUGIN(QICOPlugin)
 #include <cstdio>
 #endif // DISABLE_GUI
 
-#ifdef STACKTRACE
-#ifdef Q_OS_UNIX
-#include "stacktrace.h"
-#else
-#include "stacktrace_win.h"
-#ifndef DISABLE_GUI
-#include "stacktracedialog.h"
-#endif // DISABLE_GUI
-#endif // Q_OS_UNIX
-#endif //STACKTRACE
-
 #include "base/preferences.h"
 #include "base/profile.h"
 #include "base/version.h"
@@ -89,9 +78,6 @@ Q_IMPORT_PLUGIN(QICOPlugin)
 
 // Signal handlers
 void sigNormalHandler(int signum);
-#ifdef STACKTRACE
-void sigAbnormalHandler(int signum);
-#endif
 // sys_signame[] is only defined in BSD
 const char *const sysSigName[] =
 {
@@ -302,10 +288,6 @@ int main(int argc, char *argv[])
 
         signal(SIGINT, sigNormalHandler);
         signal(SIGTERM, sigNormalHandler);
-#ifdef STACKTRACE
-        signal(SIGABRT, sigAbnormalHandler);
-        signal(SIGSEGV, sigAbnormalHandler);
-#endif
 
         return app->exec(params.paramList());
     }
@@ -347,32 +329,6 @@ void sigNormalHandler(int signum)
     qApp->exit();  // unsafe, but exit anyway
 }
 
-#ifdef STACKTRACE
-void sigAbnormalHandler(int signum)
-{
-    const char *sigName = sysSigName[signum];
-#if !(defined Q_OS_WIN && !defined DISABLE_GUI) && !defined Q_OS_HAIKU
-    const char msg[] = "\n\n*************************************************************\n"
-        "Please file a bug report at http://bug.qbittorrent.org and provide the following information:\n\n"
-        "qBittorrent version: " QBT_VERSION "\n\n"
-        "Caught signal: ";
-    reportToUser(msg);
-    reportToUser(sigName);
-    reportToUser("\n");
-    print_stacktrace();  // unsafe
-#endif
-
-#if defined Q_OS_WIN && !defined DISABLE_GUI
-    StacktraceDialog dlg;  // unsafe
-    dlg.setStacktraceString(QLatin1String(sigName), straceWin::getBacktrace());
-    dlg.exec();
-#endif
-
-    signal(signum, SIG_DFL);
-    raise(signum);
-}
-#endif // STACKTRACE
-
 #if !defined(DISABLE_GUI)
 void showSplashScreen()
 {
