diff --git a/nvme-print.c b/nvme-print.c
index 170e5c1..cabdb95 100644
--- a/nvme-print.c
+++ b/nvme-print.c
@@ -802,6 +802,9 @@ static bool is_temperature_fahrenheit(void)
 	const char *locale, *underscore;
 	char country[3] = { 0 };
 
+#ifndef LC_MEASUREMENT
+#define LC_MEASUREMENT LC_ALL
+#endif
 	setlocale(LC_MEASUREMENT, "");
 	locale = setlocale(LC_MEASUREMENT, NULL);
 
diff --git a/plugins/netapp/netapp-nvme.c b/plugins/netapp/netapp-nvme.c
index 5c2e3d4..9f05ffc 100644
--- a/plugins/netapp/netapp-nvme.c
+++ b/plugins/netapp/netapp-nvme.c
@@ -22,6 +22,7 @@
 #include <unistd.h>
 #include <errno.h>
 #include <string.h>
+#include <libgen.h>
 
 #include "common.h"
 #include "nvme.h"
