diff --git a/nvme-rpmb.c b/nvme-rpmb.c
index 0e11ef3..2d4e069 100644
--- a/nvme-rpmb.c
+++ b/nvme-rpmb.c
@@ -23,6 +23,7 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <sys/socket.h>
+#include <limits.h>
 #include <linux/if_alg.h>
 #include <linux/socket.h>
 
diff --git a/nvme-topology.c b/nvme-topology.c
index 31cf7f9..051aaf5 100644
--- a/nvme-topology.c
+++ b/nvme-topology.c
@@ -3,6 +3,7 @@
 #include <errno.h>
 #include <unistd.h>
 #include <fcntl.h>
+#include <limits.h>
 #include <sys/stat.h>
 #include <sys/mman.h>
 
