diff --git a/gcr/console-interaction.c b/gcr/console-interaction.c
index 47c2cb4..7cedabb 100644
--- a/gcr/console-interaction.c
+++ b/gcr/console-interaction.c
@@ -62,6 +62,7 @@ GType                  console_interaction_get_type    (void) G_GNUC_CONST;
 
 G_DEFINE_TYPE (ConsoleInteraction, console_interaction, G_TYPE_TLS_INTERACTION);
 
+char *getpass(const char *);
 #ifdef G_OS_WIN32
 /* win32 doesn't have getpass() */
 static gchar *
